CSharpMath.CoreTests/LaTeXParserTest.cs

```diff
@@ -951,39 +1023,152 @@ yield return
     public void TestCustom() {
       var input = @"\lcm(a,b)";
       var builder = new LaTeXParser(input);
-      var list = builder.Build();
+      var (list, error) = builder.Build();
       Assert.Null(list);
-      Assert.NotNull(builder.Error);
+      Assert.Equal(@"Invalid command \lcm", error);
+
+      LaTeXSettings.CommandSymbols.Add(@"\lcm", new LargeOperator("lcm", false));
+      list = ParseLaTeX(input);
+      Assert.Collection(list,
+        CheckAtom<LargeOperator>("lcm"),
+        CheckAtom<Open>("("),
+        CheckAtom<Variable>("a"),
+        CheckAtom<Punctuation>(","),
+        CheckAtom<Variable>("b"),
+        CheckAtom<Close>(")")
+      );
+      Assert.Equal(@"\lcm (a,b)", LaTeXParser.MathListToLaTeX(list).ToString());
+
+      LaTeXSettings.CommandSymbols.Add(@"lcm", new LargeOperator("lcm", false));
+      LaTeXSettings.CommandSymbols.Add(@"lcm12", new LargeOperator("lcm12", false));
+      LaTeXSettings.CommandSymbols.Add(@"lcm1234", new LargeOperator("lcm1234", false));
+      LaTeXSettings.CommandSymbols.Add(@"lcm1235", new LargeOperator("lcm1235", false));
 
-      LaTeXSettings.Commands.Add("lcm", new LargeOperator("lcm", false));
-      var list2 = ParseLaTeX(input);
-      Assert.Collection(list2,
+      // Does not match custom atoms added above
+      list = ParseLaTeX("lc(a,b)");
+      Assert.Collection(list,
+        CheckAtom<Variable>("l"),
+        CheckAtom<Variable>("c"),
+        CheckAtom<Open>("("),
+        CheckAtom<Variable>("a"),
+        CheckAtom<Punctuation>(","),
+        CheckAtom<Variable>("b"),
+        CheckAtom<Close>(")")
+      );
+      Assert.Equal(@"lc(a,b)", LaTeXParser.MathListToLaTeX(list).ToString());
+
+      // Baseline for lookup as a non-command (not starting with \)
+      list = ParseLaTeX("lcm(a,b)");
+      Assert.Collection(list,
         CheckAtom<LargeOperator>("lcm"),
         CheckAtom<Open>("("),
         CheckAtom<Variable>("a"),
         CheckAtom<Punctuation>(","),
         CheckAtom<Variable>("b"),
         CheckAtom<Close>(")")
       );
-      Assert.Equal(@"\lcm (a,b)", LaTeXParser.MathListToLaTeX(list2).ToString());
+      Assert.Equal(@"\lcm (a,b)", LaTeXParser.MathListToLaTeX(list).ToString());
+
+      // We are testing a trie with nodes [l] -> l[cm] -> lcm[12] -> @lcm12[3] -> lcm123[4]
```

Comment:
Since `trie` is removed I think it would be better to rephrase comments

Reply:
Done